satellite/overlay/service.go

```diff
@@ -342,7 +342,7 @@ func (service *Service) Put(ctx context.Context, nodeID storj.NodeID, value pb.N
 	}
 
 	// Resolve IP Address Network to ensure it is set
-	value.LastIp, err = GetNetwork(ctx, value.Address.Address)
+	value.Address.Address, value.LastIp, err = ResolveAddressAndNetwork(ctx, value.Address.Address)
```

Comment:
This line will put in the IP address into the DB and that can lead to requests problems to the SN like audits and such.

Reply:
that is the point of this PR. i explained the rationale in the PR message

Reply:
This is not fullfilled? 
`So this PR is Solution B. This change persists the existing DNS resolution as the node's official address.`

Reply:
DNS resolution == ip address

Reply:
Just curious why audit and repair can't use ip addresses?

Reply:
Ok, after reading through this again, my understanding is that the ip addresses may get out of date so the satellite would not able to contact some nodes during repair and audit, which would cause audit and repair failure but that would be unfair for those nodes. Does that sound right?